lib/generators/install/templates/microservices_engine.rb

```diff
@@ -1,34 +1,69 @@
 # frozen_string_literal: true
 require 'net/http'
-CONFIG_FILE = 'config/mse_router_info.yml'
 
-unless ENV['DISABLE_ROUTER_CHECKIN']
+module MicroservicesEngine
+  class Initializer
+    CONFIG_FILE = 'config/mse_router_info.yml'
 
-  report_missing_config_key = lambda do |key|
-    raise ArgumentError, "Missing key #{key} in #{CONFIG_FILE}"
-  end
-
-  unless File.file?(CONFIG_FILE)
-    raise IOError, 'No router configuration YAML file found'
-  end
+    class << self
+      def initialize!
+        unless ENV.key? 'DISABLE_ROUTER_CHECKIN'
+          check_for_config_file! CONFIG_FILE
+          config_data = YAML.load_file CONFIG_FILE
+          response = check_in_with_router using: config_data
+          raise 'The router API response was invalid' if response.code != '200'
+          response_data = JSON.parse response.body
+          update_connections_database with: response_data
+        end
+      end
 
-  config_data = YAML.load_file(CONFIG_FILE)
+      def check_for_config_file!(filename)
+        unless File.file? filename
+          raise IOError, "No router configuration file found at #{filename}"
+        end
+      end
 
-  service_name = config_data.fetch 'name', &report_missing_config_key
-  service_url = config_data.fetch 'uri', &report_missing_config_key
-  router_url = config_data.fetch 'router_uri', &report_missing_config_key
-  security_token = config_data.fetch 'security_token', &report_missing_config_key
+      def check_in_with_router(options = {})
+        if options.key? :using
```

Comment:
I'm not that attached to it, just a stylistic choice I made to help indicate that the config data is going through some interpolation before getting passed to the router.